[DOCS] Osquery features in 8.5 (#2561)
elastic/security-docs: docs/detections/view-osquery-results.asciidoc

[[view-osquery-results]]
=== Examine Osquery results

> Osquery provides you with relevant, timely data that you can use to better understand and monitor your environment. When you run queries, results are indexed and then displayed the Results table, which you can filter, sort, and interact with.

Reviewer: ```suggestion
Osquery provides relevant, timely data that you can use to better understand and monitor your environment. When you run queries, results are indexed and displayed the Results table, which you can filter, sort, and interact with.
```
Also a little skeptical of "relevant" since that seems subjective and maybe promising too much. "Timely" is also a relative concept, but I think it's OK here.

[float]
[[osquery-result-types]]
=== Results table
The Results table displays results from single queries and query packs.

[float]
[[review-single-osquery-results]]
==== Single query results

Results for single queries appear on the *Results* tab. When you run a query, the number of agents queried and query status temporarily display in a status bar above the results table. Agent responses can be `Sucessful`, `Not yet responded` (pending), and `Failed`.

[role="screenshot"]
image::images/single-query-results.png[width=80%][height=80%][Shows query results]

[float]
[[review-pack-osquery-results]]
==== Query pack results

> Results for each query in the pack appear in the *Results* tab. Click the expand button (image:images/pack-expand-button-osquery.png[Click markdown icon,20,20]) at the far right of each query row to display query results. The number of agents that were queried and their responses are shown for each query. Agent responses are color-coded. Green is `Sucessful`, `Not yet responded` (pending) is gray, and `Failed` is red.

Reviewer: ```suggestion
Results for each query in the pack appear in the *Results* tab. Click the expand icon (image:images/pack-expand-button-osquery.png[Click markdown icon,20,20]) at the far right of each query row to display query results. The number of agents that were queried and their responses are shown for each query. Agent responses are color-coded. Green is `Sucessful`, `Not yet responded` (pending) is gray, and `Failed` is red.
```

[role="screenshot"]
image::images/pack-query-results.png[width=80%][height=80%][Shows query results]

[float]
[[investigate-osquery-results]]
=== Investigate query results

From the results table, you can:

> * Click the *View in Discover* button (image:images/discover-button-osquery.png[Click the View in Discover button,20,20]) to explore the results in Discover.
> * Click the *View in Lens* button (image:images/lens-button-osquery.png[Click the View in Lens button,20,20]) to navigate to Lens, where you can use the drag-and-drop *Lens* editor to create visualizations.
> * Click the *Timeline* button (image:images/timeline-button-osquery.png[Click Timeline button,20,20]) to investigate a single query result in Timeline or *Add to timeline investigation* to investigate all results. This option is only available for single query results.

Reviewer: Seems unnecessary & repetitive to call these out as buttons. This seems to work just as well:
```suggestion
* Click *View in Discover* (image:images/discover-button-osquery.png[Click the View in Discover button,20,20]) to explore the results in Discover.
* Click *View in Lens* (image:images/lens-button-osquery.png[Click the View in Lens button,20,20]) to navigate to Lens, where you can use the drag-and-drop *Lens* editor to create visualizations.
* Click *Timeline* (image:images/timeline-button-osquery.png[Click Timeline button,20,20]) to investigate a single query result in Timeline or *Add to timeline investigation* to investigate all results. This option is only available for single query results.
```

+
When you open all results in Timeline, the events in Timeline are filtered based on the `action_ID` generated by the Osquery query.
+

> * Click the *Add to Case* button (image:images/case-button-osquery.png[Click Add to Case button,20,20]) to add the query results to a new or existing case.

Reviewer: ```suggestion
* Click *Add to Case* (image:images/case-button-osquery.png[Click Add to Case button,20,20]) to add the query results to a new or existing case.
```

+

> NOTE: If you add the results to a new case, you are prompted to specify the solution that you want the create the case within. Ensure you select the correct solution. From {elastic-sec}, you cannot access cases created in {observability} or Stack Management.

Reviewer 2: It's so great that you mentioned these :elastic-heart:, however I think it is actually possible to access Obs and Stack cases within Security. There is no restriction about Cases solution here.

Author: Thanks for flagging this, @tomsonpl ! I re-tested this workflow and discovered two things that likely need to be noted here:
- If users create a **_new_** case, the case will only be accessible from the Solution they created the case for.
- If users are opting to add query results to an **_existing_** case, they can choose from existing cases that were made in any solution (Security, Observability, and Stack).

I'll work these in as sub-options to the step.

From the results table, you can:

* Click *View in Discover* (image:images/discover-button-osquery.png[Click the View in Discover button,20,20]) to explore the results in Discover.
* Click *View in Lens* (image:images/lens-button-osquery.png[Click the View in Lens button,20,20]) to navigate to Lens, where you can use the drag-and-drop *Lens* editor to create visualizations.
* Click *Timeline* (image:images/timeline-button-osquery.png[Click Timeline button,20,20]) to investigate a single query result in Timeline or *Add to timeline investigation* to investigate all results. This option is only available for single query results.

+
When you open all results in Timeline, the events in Timeline are filtered based on the `action_ID` generated by the Osquery query.
+

* Click *Add to Case* (image:images/case-button-osquery.png[Click Add to Case button,20,20]) to add the query results to a new or existing case.
+
[NOTE]
=====

If you add the results to a _new_ case, you are prompted to specify the solution that you want the create the case within. Ensure you select the correct solution. From {elastic-sec}, you cannot access cases created in {observability} or Stack Management.

> If you add the results to an _existing case_, you can select from cases that were created in any solution (Security, Observability, and Stack).

Reviewer 3: ```suggestion
If you add the results to an _existing case_, you can select from cases that were created in any solution ({elastic-sec}, {observability}, and {stack}).
```
